ldk/csharp/OliveHelpsLDK/SelfTestLoop/TestBuilder.cs

```diff
@@ -0,0 +1,59 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text.Json;
+using OliveHelpsLDK;
+using OliveHelpsLDK.Logging;
+using SelfTestLoop.Tests;
+
+namespace SelfTestLoop
+{
+    public class TestBuilder
+    {
+        public ILogger Logger { get; set; }
+
+        public TestBuilder(ILogger logger)
+        {
+            Logger = logger;
+        }
+
+        public ITest[] BuildTests(ILoopServices services)
+        {
+            var testServices = new TestServices()
+            {
+                Logger = Logger,
+                Services = services,
+            };
+            var testClass = new WhisperTests() {Services = testServices};
+            return GetAttributes(new[] {testClass}).Select(x => x as ITest).ToArray();
+        }
+
+        private IList<TestInstance> GetAttributes(object[] testClasses)
+        {
+            var attributes = testClasses.SelectMany(testFixture =>
+                {
+                    var methods = testFixture.GetType().GetMethods();
+                    return methods.Select(memberInfo =>
+                    {
+                        var attribute = memberInfo.GetCustomAttribute(typeof(TestAttribute));
+                        var testAttribute = attribute as TestAttribute;
+                        if (testAttribute != null)
+                            return new TestInstance()
```

Comment:
{ } - are recommended here. It will reduce a chance of future issues and increase readability.